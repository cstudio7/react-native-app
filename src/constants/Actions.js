export default {
  favoriteName: 'FAVORITE_NAME',
  unfavoriteName: 'UNFAVORITE_NAME',
  saveNames: 'SAVE_NAMES'
};