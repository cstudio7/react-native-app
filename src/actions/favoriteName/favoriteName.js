import Actions from '../../constants/Actions';

export default (gender, name) => ({
  type: Actions.favoriteName,
  gender,
  name
});
