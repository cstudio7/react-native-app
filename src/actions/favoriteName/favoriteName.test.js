import favoriteName from './favoriteName';
import Actions from '../../constants/Actions';

const mockedName = Object.freeze({ name: 'artur', isFavoite: true });
const mockedNames = Object.freeze(new Array(mockedName));

it('creates a FAVORITE_NAME action', () => {
  expect(favoriteName('male', mockedName)).toEqual({
    type: Actions.favoriteName,
    gender: 'male',
    name: mockedName
  });
  expect(favoriteName('male', mockedName)).toMatchSnapshot();
});
