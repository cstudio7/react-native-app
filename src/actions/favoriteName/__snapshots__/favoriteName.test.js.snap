// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a FAVORITE_NAME action 1`] = `
Object {
  "gender": "male",
  "name": Object {
    "isFavoite": true,
    "name": "artur",
  },
  "type": "FAVORITE_NAME",
}
`;
